File: web/src/types/article.ts
export type ArticleSource = 'manual' | 'vk' | 'telegram' | 'habr'

export interface Category {
  id: number
  title: string
  created_at: string
  updated_at?: string
}

export interface Article {
  id: number
  title: string
  body: string
  url: string
  source: ArticleSource
  source_id?: string
  categories: Category[]
  created_at: string
  updated_at?: string
}

export interface CreateArticleRequest {
  title: string
  body: string
  url: string
  source: ArticleSource
  source_id?: string
  categories: string[]
}

export interface UpdateArticleRequest {
  title?: string
  body?: string
  url?: string
  categories?: string[]
}

export interface ArticleFilter {
  source?: ArticleSource
  category_id?: number
  search?: string
  limit?: number
  offset?: number
}

export interface ListResponse<T> {
  data: T[]
  total: number
  limit: number
  offset: number
}

File: internal/dto/article.go
package dto

import "time"

type ArticleSource string

const (
	SourceManual   ArticleSource = "manual"
	SourceVK       ArticleSource = "vk"
	SourceTelegram ArticleSource = "telegram"
	SourceHabr     ArticleSource = "habr"
)

type Article struct {
	ID         int           `json:"id"`
	Title      string        `json:"title"`
	Body       string        `json:"body"`
	URL        string        `json:"url"`
	Source     ArticleSource `json:"source"`
	SourceID   *string       `json:"source_id,omitempty"`
	Categories []Category    `json:"categories"`
	CreatedAt  time.Time     `json:"created_at"`
	UpdatedAt  *time.Time    `json:"updated_at,omitempty"`
}

type CreateArticleRequest struct {
	Title      string        `json:"title"`
	Body       string        `json:"body"`
	URL        string        `json:"url"`
	Source     ArticleSource `json:"source"`
	SourceID   *string       `json:"source_id,omitempty"`
	Categories []string      `json:"categories"`
}

type UpdateArticleRequest struct {
	Title      *string  `json:"title,omitempty"`
	Body       *string  `json:"body,omitempty"`
	URL        *string  `json:"url,omitempty"`
	Categories []string `json:"categories,omitempty"`
}

type ArticleFilter struct {
	Source     *ArticleSource
	CategoryID *int
	Search     *string
	Limit      int
	Offset     int
}
